### PR TITLE
fix(storefront): BCTHEME-1897 Date Field Modifier - February showing 30th and 31st

### PR DESCRIPTION
**Pull request by bc-vlad-dlogush**
Jira: [BCTHEME-1870](https://bigcommercecloud.atlassian.net/browse/BCTHEME-1897)

## What/Why?
This PR fixes an issue with displaying the correct number of days when a specific date is selected

## Rollout/Rollback
No Rollout/Rollback

## Testing

https://github.com/user-attachments/assets/b496f6de-91b1-4a04-ab53-fe10994b19d4

https://github.com/user-attachments/assets/5b4a901a-1c9e-4fdd-8751-c0aad55008c1

[BCTHEME-1870]: https://bigcommercecloud.atlassian.net/browse/BCTHEME-1870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by BC-krasnoshapka — APPROVED** [review]

File: assets/js/theme/common/product-details.js [hunk 1]
```diff
@@ -591,4 +592,41 @@ export default class ProductDetails extends ProductDetailsBase {
...
+
+            const updateDays = () => {
+                const month = parseInt(monthSelector.val(), 10);
+                const year = parseInt(yearSelector.val(), 10);
+                let daysInMonth;
+
+                if (!Number.isNaN(month) && !Number.isNaN(year)) {
+                    switch (month) {
+                    case 2:
+                        daysInMonth = (year % 4 === 0 && year % 100 !== 0) || (year % 400 === 0) ? 29 : 28;
```
Comment (bc-alexsaiannyi) [inline assets/js/theme/common/product-details.js:610]: @VladDlogush as I understand this line we basically need to check a leap year and if so return 29days for February. What's the reason behind  checking for `year % 100` or `year % 400` ?

File: assets/js/theme/common/product-details.js [hunk 2]
```diff
@@ -591,4 +592,41 @@ export default class ProductDetails extends ProductDetailsBase {
             detail: { productDetails },
         }));
     }
+
+    updateData() {
```
Comment (bc-alexsaiannyi) [inline assets/js/theme/common/product-details.js:596]: as a suggestion here, maybe to give it a more specific name, like `updateDateOption` or `updateDatePickerOption`, `updateDateSelector`?🤔

File: assets/js/theme/common/product-details.js (hunk 1 above)
Comment (yurytut1993) [reply assets/js/theme/common/product-details.js:610]: @bc-alexsaiannyi this are rules how leap year can be defined. Leap year - this year it is included in 4, but not included in 100 unless it is included in 400

**Review by yurytut1993 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Draft
+- Date Field Modifier - February showing 30th and 31st [#2472](https://github.com/bigcommerce/cornerstone/pull/2472)
```
Comment (bc-yevhenii-buliuk) [inline CHANGELOG.md:8]: @bc-vlad-dlogush fix the pr's number please

**Review by bc-yevhenii-buliuk — APPROVED** [review]

File: assets/js/theme/common/product-details.js (hunk 1 above)
Comment (bc-vlad-dlogush) [reply assets/js/theme/common/product-details.js:610]: @yurytut1993 is right! This is the rule for determining a leap year
@bc-alexsaiannyi

File: CHANGELOG.md (hunk 3 above)
Comment (bc-vlad-dlogush) [reply CHANGELOG.md]: Thanks!

File: assets/js/theme/common/product-details.js (hunk 2 above)
Comment (bc-vlad-dlogush) [reply assets/js/theme/common/product-details.js]: I agree. `updateDateSelector` is suitable in my opinion.

**Review by bc-alexsaiannyi — APPROVED** [review]
👌

**Review by bc-azvierieva — APPROVED** [review]

**Pull request merged**